[syj02] iP

syj02 opened the pull request.

Commit:
style: Rename chatbot

Commit:
feat: Add greetings and exit
Implement initial greeting and exiting messages

Commit:
feat: Add echo
Chatbot Jane replies by echoing the user's input

Commit:
feat: Add store and list
Implement arrays such that chatbot has the ability to store whatever text entered by the user and display them back to the user when requested

Commit:
refactor: Abstract task into another class
Created a new class Task that contains the description of String type and status of Boolean type to keep the individual tasks

Commit:
refractor: Abstrac task list into another class
Created TaskList to store all the tasks in a list and keep track of the total number of tasks added

Commit:
feat: Add mark and unmark functionalities
Added mark and unmark to update whether the tasks are done.

Commit:
style: Modify code to fit coding standards
Modifications made:
*if-else statements with correct indentations
*System.out.println statements with suitable length per line

Commit:
feat: Modify isDone method
Modified isDone method in Task class with clearer parameters:
Changed parameter from String to Boolean for shorter method

Commit:
style: Modify code to fit coding standards
Align indentations for logo for better readability
Rename the following to constants convention:
* logo to LOGO
* horizontalLine to SEPARATOR
* greetMessage to GREET_MESSAGE
* exitMessage to EXIT_MESSAGE
Name change to SEPARATOR for better readability

Commit:
style: Rename variable in Task and TaskList
Rename index to sequenceNumber as the variable represents the addition sequence of tasks.

Commit:
Revert "style: Rename variable in Task and TaskList"

This reverts commit 7b1a86db6c03514032fa21ac7fb32a4775a6c3b8.

Commit:
style: Rename variable in Task
Rename index to sequence as it represents the addition sequence of tasks in task list.
Update methods in Jane, Task, and TaskList to match the new name:
* Change all index to sequence
* Change getIndex() change to getSequence() in Task

Commit:
Revert "style: Rename variable in Task"

This reverts commit 663399b99c714c01a66862f1c71d1b1933fd47a3.

Commit:
style: Rename variable in Task
Rename index to sequence as it represents the addition sequence of tasks in task list.
Update methods in Jane, Task, and TaskList to match the new name:
* Change all index to sequence
* Change getIndex() change to getSequence() in Task

Commit:
refactor: Add method to process input
Add new method processInput(String, TaskList)
In new method:
* Add new variable String[] inputPart to split input into two parts to know the action required (i.e. "list", "mark", "unmark")
* Change if-else statements to switch case for better readability and shorter code

Commit:
fix: Remove static final modifiers from variables
Remove static final modifiers from SEPARATOR, GREET_MESSAGE, EXIT_MESSAGE

Commit:
refactor: Add separator
Add print SEPARATOR for output with better readability

Commit:
refactor: Abstract new classes
Add new subclasses of Task class for multiple task types
* Deadline class for tasks to be done before specific date/time
* Event class for tasks with specific start date/time and end date/time
* Todo class for tasks to be done without date/time limit

Commit:
fix: Modify inheritance of Event
Add extends Task to add inheritance from Task class

Commit:
feat: Add toString() method
Override default toString() method to return status icon and description of task

Commit:
feat: Add constructor and toString() method
* Add constructor which inherits from Task constructor
* Add toString() method inherit from Task and override with addition of task type icon

Commit:
feat: Add constructor and toString() method
* Add constructor which inherits from Task constructor and the date by which task needs to be completed
* Add toString() method inherit from Task and override with addition of task type icon and the date by which task needs to be completed

Commit:
feat: Add constructor and toString() method
* Add constructor which inherits from Task constructor and add start and end date/time for task
* Add toString() method inherit from Task and override with addition of task type icon and the start and end date/time for task

Commit:
refactor: Modify toString() method
Modify toString() method for the printed message with better readability

Commit:
refactor: Modify toString() method in Event class
Modify toString() to have the print out message that follows the sample given

Commit:
feat: Add getCount() method in TaskList class
Add getCount() method to access the counts for number of tasks in taskList

Commit:
feat: Modify processInput() method in Jane class
Modify the processInput() method:
* Add case "todo" in switch case for todo task
* Add case "deadline" in switch case for deadline task
* Add case "event" in switch case for event task
* Modify print out message in case "mark" and make use of toString() method
* Modify print out message in case "unmark" and make use of toString() method
* Modify default to print correct usage message that have the wrong input message

XavierLiau34 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,15 @@
+public class Event extends Task {
+    protected String start;
+    protected String end;
+    public Event(String description, String start, String end) {
```
Please add a line break/new line between line 3 and 4

EdmundTangg replied:
Can add an empty line in between variable declarations and function for better visibility.

XavierLiau34 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,12 @@
+public class Deadline extends Task {
+    protected String by;
+    public Deadline(String description, String by) {
```
Please add a line break/new line between line 2 and 3

XavierLiau34 commented on src/main/java/Jane.java:
```diff
@@ -0,0 +1,82 @@
+    public static void processInput(String input, TaskList taskList) {
+        String[] inputPart = input.split(" ", 2);
+        Task currentTask;
+        switch (inputPart[0]) {
```
Please consider making this shorter if possible.

EdmundTangg replied:
Good usage of switch cases.

EdmundTangg commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,23 @@
+public class TaskList {
```
Good usage of TaskList class to add the tasks into the list array.

XavierLiau34 commented on src/main/java/Jane.java:
```diff
@@ -0,0 +1,82 @@
+            break;
+        }
+    }
+    public static void main(String[] args) {
```
Please fix the indentation for line 54.

XavierLiau34 commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,10 @@
+    public Todo(String description) {
+        super(description);
+    }
+    @Override
```
Please add a line break/new line between lines 4 and 5, and remove the new line on line 9

EdmundTangg commented on src/main/java/Jane.java:
```diff
@@ -0,0 +1,82 @@
+                    "Now you have " + taskList.getCount() + " tasks in the list.");
+            break;
+        case "deadline":
+            String[] deadlineInput = inputPart[1].split("/", 2);
```
might be better if the variables declaration are all grouped together before the switch case starts.

XavierLiau34 reviewed (commented):
Overall, LGTM:), only a little nits. Your code is great, don't worry too much about it.

EdmundTangg reviewed (commented):
LGTM. Just some minor details to take note of. Well done!

Commit:
test: Modify scripts to test Jane
* Modified the input.txt file by adding the essential calls to test the 3 task types, mark, unmark and list
* Modified the EXPECTED.txt file by adding in the expected output
* Modified the runtest.sh file to call Jane instead Duke due to name change of chatbot main file
* Modified the runtest.bat file to call Jane instead Duke due to name change of chatbot main file
* The ACTUAL.txt file is the output auto-generated by running Jane with inputs from input.txt

Commit:
style: Add and Remove empty lines of code
Add and Remove empty lines within and between methods for better readability and consistency of addition of empty lines

Commit:
refactor: Abstract process input methods into TaskList
Modify processInput() in Jane to shorten the code and call methods in TaskList to process the inputs instead
* Add printList() to print full task list
* Add markAsDone() to mark task as done
* Add markAsUndone() to mark task as undone
* Add processTodo() to add todo task and print out the number of tasks in task list
* Add processDeadline() to add deadline task and print out the number of tasks in task list
* Add processEvent() to add event task and print out the number of tasks in task list

Commit:
refactor: Modify indentations to fit coding convention

Commit:
refactor: Add JaneException class

Commit:
feat: Handle exceptions in Jane

Commit:
style: Remove empty line in Jane

Commit:
Reset branch commit before merge of 'master' branch

Commit:
Merge branch 'branch-Level-5'

Commit:
feat: Add delete method using collections
Modify code to use ArrayList<> for taskList
Add delete method in taskList to remove task

Commit:
feat: Add storage to hard disk

Commit:
Merge branch 'branch-Level-6'

Commit:
Merge branch 'branch-Level-7'

Commit:
fix: Bug fix for save and load files
Modify code to fit the standard save format and to read the txt file

json-bot-boop commented on src/main/java/Jane.java:
```diff
@@ -0,0 +1,96 @@
+import java.io.IOException;
+import java.util.Scanner;
+public class Jane {
+public static final String FILE_PATH = "./data/jane.txt";
+    public static Storage storage;
```
There is no indentation for line 4

json-bot-boop commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,17 @@
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by + ")";
```
Can considering using the getters since you have already created them

json-bot-boop commented on src/main/java/Jane.java:
```diff
@@ -0,0 +1,96 @@
+            String[] inputPart = input.split(" ", 2);
+            if (inputPart.length < 2 || inputPart[1] == null) {
+                switch (inputPart[0]) {
+                case "todo" :
```
Try to avoid magic numbers

json-bot-boop commented on src/main/java/Jane.java:
```diff
@@ -0,0 +1,96 @@
+                break;
+            default:
+                throw new JaneException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+            }
```
Try to avoid magic string

json-bot-boop commented on src/main/java/Jane.java:
```diff
@@ -0,0 +1,96 @@
+    }
+
+    public static void main(String[] args) {
+        String LOGO = " _____    _____    ____ _    _____ \n"
+                + "|____ |  |     |  |    | |  | ____|\n"
+                + "    | |  |  |  |  | |  | |  | |___ \n"
+                + " _  | |  |  _  |  | |  | |  |  ___|\n"
+                + "| |_| |  | | | |  | |  | |  | |___ \n"
+                + "|_____|  |_| |_|  |_| ___|  |_____|\n";
+        String SEPARATOR = "____________________________________________________________\n";
+        String GREET_MESSAGE = "Hello! I am Jane.\nWhat can I do for you?\n";
+        String EXIT_MESSAGE = "Bye. Hope to see you again soon!\n";
```
The variables are not in constant but are in screaming snake case

json-bot-boop commented on src/main/java/Jane.java:
```diff
@@ -0,0 +1,96 @@
+            Scanner in = new Scanner(System.in);
+            String input = in.nextLine();
+
+            while (!input.equals("bye")) {
+                try {
+                    System.out.print(SEPARATOR);
+                    processInput(input, taskList);
+                    System.out.print(SEPARATOR);
+                    storage.saveTasksToFile(taskList);
+                } catch (JaneException e) {
+                    System.out.println(e.getMessage());
+                    System.out.print(SEPARATOR);
+                }
+                input = in.nextLine();
+            }
```
Try to abstract your methods to avoid arrowhead code

json-bot-boop reviewed (commented):
Overall, the code looks good except some code quality and coding standard issues

Commit:
refactor: Abstract ui methods into Ui
Create new class Ui to deal with user interactions
Add nextInput method to read in next line of input from user

Commit:
refactor: Abstract process input method into Ui

Commit:
refactor: Abstract methods into Parser
Create Parser to make sense of user inputs

Commit:
style: Modify code to fit coding standards
* Remove extra newlines at the end across classes
* Remove magic number and add constant in Jane
* Modify naming of variables in Jane

Commit:
style: Remove extra white space

Commit:
feat: Add findWord method
Method to search for tasks in task list that contains keyword input by user

Commit:
docs: Add JavaDoc documentation for all existing classes

syj02 committed:
Merge pull request #1 from syj02/branch-Level-9

branch Level 9

syj02 committed:
Merge pull request #2 from syj02/A-JavaDoc

A java doc

Commit:
refactor: Make use of exisiting get methods
Modify Deadline and Event classes to use getters

Commit:
refactor: Remove unused import in Jane

Commit:
refactor: Remove unused variable count in TaskList
Remove unused variable count and its corresponding documentation

Commit:
style: Edit newlines in class
Add or Remove newlines for better readability and standardization across classes

Commit:
docs: Add missing documentation for constructor

Commit:
docs: Modify documentation in Deadline
* Add missing documentation
* Modify wrongly written documentations

Commit:
style: Reorder methods for better readability and logical structure

Commit:
Add a User Guide

Commit:
Udpate User Guide.

Commit:
refactor: Abstract all messages into Message
Abstract all the messages printed that are magic strings into Message.
Some modifications of messages done to provide clarity.

Commit:
refactor: Add static to constant EXPECTED_LENGTH

Commit:
style: Modify Separator and Greet format
* Increase length of separator
* Add Separator before printing logo

Commit:
fix: Remove additional bracket

Commit:
refactor: Rename EXPECTED_LENGTH variable
Rename variable to EXPECTED_INPUT_LENGTH for clarity due to subsequent similar variables added

Commit:
refactor: Add 2 variables for the expected length for deadline and event

Commit:
feat: Add new exceptions for processDeadline and processEvent
Add exception to handle inputs with missing information

Commit:
feat: Add new exceptions for methods that require integer input
Modify markAsDone, markAsUndone, and removeTask methods:
* Add exception to handle input that cannot be converted to integer
* Add exception to handle input with index out of bounds
* Add the respective JavaDoc documentation

Commit:
feat: Add messages for new exceptions handled

Commit:
fix: Modify the parameter names provided

Commit:
style: Rename output data file to fit coding conventions

Commit:
chore: Add Jane.txt to .gitignore

Commit:
feat: Add new exception to handle missing input

Commit:
refactor: Modify exception handling message

syj02 committed:
Update README.md
